presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/LogicalCteOptimizer.java

```diff
@@ -132,34 +139,68 @@ public boolean isPlanRewritten()
         }
     }
 
-    public class CteConsumerTransformer
+    public static class CteConsumerTransformer
             extends SimplePlanRewriter<CteTransformerContext>
     {
         private final PlanNodeIdAllocator idAllocator;
 
         private final VariableAllocator variableAllocator;
 
-        public CteConsumerTransformer(PlanNodeIdAllocator idAllocator, VariableAllocator variableAllocator)
+        private final Session session;
+
+        private static final List<Class<? extends PlanNode>> PRECOMPUTE_PLAN_NODES = ImmutableList.of(JoinNode.class, SemiJoinNode.class, AggregationNode.class);
```

Comment:
should we call these two COMPLEX_PLAN_NODES and COMPLEX_DATA_SOURCE_NODES or something like that to better correlate with the session property name?